cachi2/core/package_managers/pip.py

```diff
@@ -237,6 +245,46 @@ def get_version(self) -> Optional[str]:
         """Attempt to determine the package version. Should only be called if file exists."""
 
 
+class PyProjectTOML(SetupFile):
+    """Extract project name and version from a pyproject.toml file."""
+
+    def __init__(self, top_dir: RootedPath) -> None:
+        """
+        Initialize a PyProjectTOML.
+
+        :param top_dir: Path to root of project directory
+        """
+        super().__init__(top_dir, "pyproject.toml")
+
+    def get_name(self) -> Optional[str]:
+        """Get project name if present."""
+        try:
+            return self._parsed_toml["project"]["name"]
+        except KeyError:
+            return None
+
+    def get_version(self) -> Optional[str]:
+        """Get project version if present."""
+        return self._get_statically_set_version()
+
+    def _get_dynamically_set_version(self) -> Optional[str]:
+        # TODO
```

Comment:
How would this work? Do we intend to leave this as TODO for now?

Reply:
Yeah, makes sense